src/models/simulation.ts

```diff
@@ -123,39 +86,88 @@ export class SimulationModel {
       direction: config.windDirection
     };
 
-    const landType: LandType[] | undefined =
-      config.landType && populateGridWithImage(this.gridHeight, this.gridWidth, config.landType);
-    const elevation: number[] | undefined =
-      config.elevation && populateGridWithImage(this.gridHeight, this.gridWidth, config.elevation);
-    for (let y = 0; y < this.gridHeight; y++) {
-      for (let x = 0; x < this.gridWidth; x++) {
-        const index = getGridIndexForLocation(x, y, this.gridWidth);
-        const cellOptions: CellOptions = { x, y };
-        if (landType) {
-          cellOptions.landType = landType[index];
-        }
-        if (elevation) {
-          cellOptions.elevation = elevation[index];
-        }
-        this.cells.push(new Cell(cellOptions));
-      }
-    }
     // It's enough to calculate this just once, as grid won't change.
     this.cellNeighbors = calculateCellNeighbors(this.gridWidth, this.gridHeight, this.config.neighborsDist);
-    // It's enough to calculate this just once, as long as none of the land properties or wind speed can be changed.
-    // This will change in the future when user is able to set land properties or wind speed dynamically.
-    this.timeToIgniteNeighbors = calculateTimeToIgniteNeighbors(this.cells, this.cellNeighbors, this.wind, config);
 
-    if (config.spark) {
-      const sparkX = Math.round(config.spark[0] / this.config.cellSize);
-      const sparkY = Math.round(config.spark[1] / this.config.cellSize);
-      this.cells[sparkX * this.gridWidth + sparkY].ignitionTime = 1;
-    }
+    this.populateCellsData();
+
+    // Make simulation available in browser console for manual tests.
+    (window as any).sim = this;
+  }
+
+  public getLandTypeData(): Promise<number[]> {
```

Comment:
How come this is a Promise? `populateGrid` is synchronous. AFAIK, `Promises.all` works fine if you pass in a non-promise function.

Reply:
...oh, maybe just for future-proofing for when landTypes comes in as an image. :+1:

Reply:
Yes, it's a Promise that resolves immediately. Just so it's consistent with `getElevationData` and as you said, so it's easier to load land type data from the image in the future (which will have to happen I guess).